File: lc_soln/0875_koko_eating_bananas/solution.py
import math
from typing import List


class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1
        r = max(piles)
        result = r
        while l <= r:
            mid = l + (r - l) // 2
            count = sum(math.ceil(pile / mid) for pile in piles)
            if count <= h:
                result = min(result, mid)
                r = mid - 1
            else:
                l = mid + 1
        return result


def main(): ...


if __name__ == "__main__":
    main()

File: lc_soln/0875_koko_eating_bananas/solution.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int l = 1;
        int r = *std::max_element(piles.begin(), piles.end());
        int result = r;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            int count = 0;
            for (const int pile : piles) {
                count += (pile + mid - 1) / mid;
            }
            if (count <= h) {
                result = mid;
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        return result;
    }
};

int main() {
    return 0;
}
